homeassistant/components/device_tracker/keenetic_ndms2.py

```diff
@@ -69,53 +72,35 @@ def scan_devices(self):
 
     def get_device_name(self, device):
         """Return the name of the given device or None if we don't know."""
-        filter_named = [result.name for result in self.last_results
-                        if result.mac == device]
+        filter_name = [device.name for device in self.last_results
+                       if device.mac == device]
 
-        if filter_named:
-            return filter_named[0]
+        if filter_name:
+            return filter_name[0]
         return None
 
+    def get_extra_attributes(self, device):
+        """Return the IP of the given device."""
+        filter_ip = [result.ip for result in self.last_results
```

Comment:
Use a generator expression and `next` here too.

Reply:
Is get_extra_attributes expected to return None?